uat/assets/test-job.yaml.j2

```diff
@@ -0,0 +1,37 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ job_name }}
+spec:
+  backoffLimit: 0
+  template:
+    metadata:
+      labels:
+          access-minio: "true"
+          access-ml-pipeline: "true"
+          mlflow-server-minio: "true"
+    spec:
+      serviceAccountName: default-editor
+      containers:
+        - name: {{ job_name }}
+          image: {{ test_image }}
+          command:
+            - bash
+            - -c
+          args:
+            - |
+              cd /tests;
+              pip install -r requirements.txt >/dev/null;
+              pytest;
+              # Kill Istio Sidecar after workload completes to have the Job status properly updated
+              # https://github.com/istio/istio/issues/6324
+              x=$(echo $?);
+              curl -fsI -X POST http://localhost:15020/quitquitquit >/dev/null && exit $x;
+          volumeMounts:
+            - name: test-volume
+              mountPath: /tests
+      volumes:
+        - name: test-volume
+          hostPath:
+            path: {{ test_dir }}
+      restartPolicy: Never
```

Comment:
Why is this never ? I see bellow we use tenacity to run the test multiple times if failed but we can achieve this by setting 

Detting this OnFailure and then we can set this https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-backoff-failure-policy

The advatage of this approach is that only the container will be restarted not the whole pod (as in case of tenacity solution).

Reply:
> I see bellow we use tenacity to run the test multiple times if failed

What's "below"? Where do we do that? I use tenacity only to wait for stuff to happen, I don't rerun the test multiple times, if it fails, it fails.

And that's exactly why the `restartPolicy` is set to `Never` (and the `backoffLimit` to 0). I don't want the tests to be retried. If something fails, great, we get a report about it and move on. Why would we want to retry all of tests?

Reply:
yeah if I'm understanding this correctly, I think having this set to `Never` makes sense